### PR TITLE
Cache sql statements triggered internally

### PR DESCRIPTION
**Pull request by nirbhayc**
Part-2 of there work done in #2310 
Signed-off-by: Nirbhay Choubey <nchoubey@bloomberg.net>

### CODE REVIEW

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/6c26f40f/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/6c26f40f/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/6c26f40f/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2662'>Resubmit</a></p>

File: db/sqlinterfaces.c [hunk 1]
```diff
@@ -4444,20 +4445,41 @@ static int execute_verify_indexes(struct sqlthdstate *thd,
...
+        if (thd->stmt_cache == NULL) {
+            thd->stmt_cache = stmt_cache_new(NULL);
+        }
+
+        stmt_cache_entry_t *cached_entry;
+        if ((stmt_cache_find_entry(thd->stmt_cache, clnt->sql,
+                                   &cached_entry)) == 0) {
+            stmt = cached_entry->stmt;
+        } else {
+            clnt->prep_rc = rc =
```
Comment (adizaimi) [inline db/sqlinterfaces.c:4459]: May I suggest you merge the two else-s into one?
```
found = 0;
If (gbl_enable) {
  ..
  if(stmt_find) {
     ...
     found = 1;
   }
}
if (!found) {
   prepare();
}
  ```

**Review by riverszhang89 — DISMISSED** [review]

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/87a273e4/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p>The first 10 failing tests are: <br>concurrent_sql</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/87a273e4/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/87a273e4/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2662'>Resubmit</a></p>

**Review by riverszhang89 — APPROVED** [review]

**Comment by nirbhayc** [comment]
Thanks for the review @riverszhang89 !

**Pull request merged**